src/Core/Localization/CLDR/TerritoryData.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Localization\CLDR;
+
+/**
+ * The TerritoryData class is the exact representation of Territory's data structure inside CLDR xml data files.
+ */
+class TerritoryData
+{
+    /**
+     * Alphabetic ISO 4217 territory code.
+     *
+     * @var string
+     */
+    protected $isoCode;
+
+    /**
+     * Territory code.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * Override this object's data with another TerritoryData object.
+     *
+     * @param TerritoryData $territoryData Currency data to use for the override
+     *
+     * @return $this Fluent interface
+     */
+    public function overrideWith(TerritoryData $territoryData)
+    {
+        if (null !== $territoryData->getIsoCode()) {
+            $this->setIsoCode($territoryData->getIsoCode());
+        }
+
+        if (null !== $territoryData->getName()) {
```

Comment:
it can happen to have in parameter a territoryData with empty name. In this case, the name of the current TerritoryData will be keep (kept ?), with the isoCode of the given TerritoryData

Reply:
As you can see in `src/Core/Localization/CLDR/Reader.php`, I check that the name can't be empty. But the interface accepting null values, I must check that null values are not replacing existing values like isocode.

Reply:
I think the method must check that the object given is complete to prevent weird behaviours

Reply:
The different layers are supposed to handle this correctly If it doesn't want to eras a value with empty string then it doesn't set anything This way the field is null and doesn't override anything

However some layer might legitimately wants an empty value in which case it should be able to override previous layers

When I look at LocaleData it doesn't check empty strings and it works fine for now so I think we can rely on the same architecture